articles/modules/ROOT/pages/migrating-explicit-lucene-indexes-to-native-schema-indexes.adoc

```diff
@@ -0,0 +1,102 @@
+ 
+As to the background, Legacy/explicit indexes were the only type of indexes available prior to Neo4j 3.2(release date 2017).   These early version indexes which were implemented via Lucene under the cover resulted in significant write performance, and thus the need to replace them with the native schema indexex available in 3.3+ releases.   
+
+Aside from the performance perspective and equally important, explicit indexes also have to be explicitly/manually kept-up-to date when adding/removing/updating nodes/relationships/properties by way of the Java API (and/or stored procedures starting with 3.3.x), hence why they are called an explicit index.    
+
+In contrast, native indexes are maintained automatically, thus eliminating the need for any such manual steps and/or extra code to keep them current.
+
+From an implementation perspective, these legacy indexes do not list much of schema details and besides the original author's chosen naming convention at the time of the index creation, there is not a whole lot of usable clue to help with implementing an automated process for migrating these indexes to Neo4j’s native schema indexes in a scripted manner.
+
+Thus any index migration would have to be done on an index by index basis and might involve some level of guessing.   This means you having to look at the API code and assess which node or relationship properties are being indexes, and accordingly implement a complementing schema index on those properties and/or look at the cypher query and come up with supporting indexes to optimize its execution.
+
+Overall at a high level, here is the recommended approach:
+
+1) Get a listing of all your indexes on you existing version
+
+*  Example:   in 3.x you can use **"call db.index.explicit.list();"** which will show the index name as well as its type, which can be either “exact” or “full-text”, as well as if it is a node or relationship index, which will help you to determine what type of schema index to construct. 
+ 
+2) Next look at your cypher and/or Java API code and convert them accordingly to 4.x+ format.   
+
+* For example, convert "start" statements to equivalent Match statement.  Example: 
+**  **START n=node:myExplicitIndexYear("myid:1234567") RETURN n;**
+** The above query would be changed to the following:
+**  **MATCH (n:Person{myid:1234567}) RETURN n;**
+
+ 
+
+3) Implement the equivalent schema index that you would need to support and optimize the execution of the associated/converted queries from step 2. 
+
+* Example:   **CREATE INDEX index_name FOR (n:Person) ON (n.myid);**
+* More specifically:
+** Inspect all cypher queries for what they actually write to that index and derive either a create index (for integer, strings, dates, geospatial, etc) and/or call db.index.fulltext.createIndexForNodes (FTS) statements.
```

Comment:
Note, with 4.3, FTI procs are deprecated. From 4.3.0 changelog:
"Introduce CREATE FULLTEXT INDEX... Cypher command. Deprecate db.index.fulltext.createNodeIndex, db.index.fulltext.createRelationshipIndex and db.index.fulltext.drop.". We might want to mention this as an additional option whilst highlighting the deprecation.